### PR TITLE
support validation regression test

### PR DESCRIPTION
**Pull request by OsmondJiang**
[AB#244872](https://dev.azure.com/ceapex/d3d54af3-265a-4f18-95f6-9a46397ca583/_workitems/edit/244872)

Want to use one specific ruleset for PR regression test, to cover most of the validation scenarios

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/6244)

### CODE REVIEW

File: azure-pipelines-v3.yml [hunk 1]
```diff
@@ -133,17 +133,17 @@ jobs:
       DocsRoot:
         params: https://github.com/MicrosoftDocs/DocsRoot --timeout 20
       azure-docs-pr:
-        params: https://github.com/MicrosoftDocs/azure-docs-pr --timeout 80
+        params: https://github.com/MicrosoftDocs/azure-docs-pr --timeout 80 -markdown-rules-api https://ops/regressionallcontentrules/ -metadata-schema-api https://ops/regressionallmetadataschema/
```
Comment (yufeih) [inline azure-pipelines-v3.yml:136]: Use a simpler bool switch?

File: test/docfx.RegressionTest/RegressionTest.cs [hunk 2]
```diff
@@ -90,6 +90,17 @@ static string GetDocfxConfig(Options opts)
                     docfxConfig["outputUrlType"] = "ugly";
                     docfxConfig["template"] = "https://github.com/Microsoft/templates.docs.msft.pdf#master";
                 }
+
+                if (opts.RegressionMarkdownRule)
+                {
+                    docfxConfig["markdownValidationRules"] = "https://ops/regressionallcontentrules/";
```
Comment (live1206) [inline test/docfx.RegressionTest/RegressionTest.cs:96]: why not set the actual URL directly here?

Comment (OsmondJiang) [reply test/docfx.RegressionTest/RegressionTest.cs:96]: the actual url is enabled AuthN

Comment (live1206) [reply test/docfx.RegressionTest/RegressionTest.cs:96]: ok

**Pull request merged**